### PR TITLE
build(jest): Use `jest-circus` as test runner

### PR DESCRIPTION
**Pull request by billyvg**
This is the next-gen test runner for jest. It provides an event handler so that we can add apm to our test suite.

See https://github.com/facebook/jest/tree/master/packages/jest-circus for more information.

### CODE REVIEW

File: tests/js/spec/api.spec.jsx [hunk 1]
```diff
@@ -260,24 +259,4 @@ describe('api', function() {
       );
     });
   });
-
-  describe('Sentry reporting', function() {
-    beforeEach(function() {
```
Comment (billyvg) [inline tests/js/spec/api.spec.jsx]: Empty beforeEach that does nothing.

File: tests/js/spec/views/projectPluginDetails.spec.jsx [hunk 2]
```diff
@@ -58,11 +63,6 @@ describe('ProjectPluginDetails', function() {
     );
   });
 
-  afterAll(function() {
-    // eslint-disable-next-line no-console
-    console.info.restore();
```
Comment (billyvg) [inline tests/js/spec/views/projectPluginDetails.spec.jsx]: `restore` is not a valid jest mock fn fn

**Comment by EvanPurkhiser** [comment]
Whoa this is cool af

File: .travis.yml [hunk 3]
```diff
@@ -34,7 +34,7 @@ env:
     - DJANGO_VERSION=">=1.11,<1.12"
     # node's version is pinned by .nvmrc and is autodetected by `nvm install`.
     - NODE_DIR="${HOME}/.nvm/versions/node/v$(< .nvmrc)"
-    - NODE_OPTIONS=--max-old-space-size=4096
+    - NODE_OPTIONS=--max-old-space-size=6144
```
Comment (billyvg) [inline .travis.yml]: circus seems to require for memory :(

Comment (billyvg) [reply .travis.yml]: See https://github.com/facebook/jest/issues/8816

**Comment by billyvg** [comment]
Going to gather more data before trying to get this merged.

**Comment by BYK** [comment]
Closing due to staleness and conflicts.

**Pull request closed without merging**